include/cpr/ssl_options.h

```diff
@@ -14,6 +20,324 @@ class VerifySsl {
     bool verify_ = true;
 };
 
+namespace ssl {
+
+// set SSL client certificate
+class CertFile {
+  public:
+    template <typename FileType>
+    CertFile(FileType&& p_filename) : filename(CPR_FWD(p_filename)) {}
+
+    std::string filename;
+
+    virtual const char* GetCertType(void) const {
+        return "PEM";
+    }
+};
+
+typedef CertFile PemCert;
+
+class DerCert : public CertFile {
+  public:
+    template <typename FileType>
+    DerCert(FileType&& p_filename) : CertFile(CPR_FWD(p_filename)) {}
+
+    virtual const char* GetCertType(void) const {
+        return "DER";
+    }
+};
+
+// specify private keyfile for TLS and SSL client cert
+class KeyFile {
+  public:
+    template <typename FileType>
+    KeyFile(FileType&& p_filename) : filename(CPR_FWD(p_filename)) {}
+
+    template <typename FileType, typename PassType>
+    KeyFile(FileType&& p_filename, PassType p_password)
+            : filename(CPR_FWD(p_filename)), password(CPR_FWD(p_password)) {}
+
+    std::string filename;
+    std::string password;
+
+    virtual const char* GetKeyType(void) const {
+        return "PEM";
+    }
+};
+
+typedef KeyFile PemKey;
+
+class DerKey : public KeyFile {
+  public:
+    template <typename FileType>
+    DerKey(FileType&& p_filename) : KeyFile(CPR_FWD(p_filename)) {}
+
+    template <typename FileType, typename PassType>
+    DerKey(FileType&& p_filename, PassType p_password)
+            : KeyFile(CPR_FWD(p_filename), CPR_FWD(p_password)) {}
+
+    virtual const char* GetKeyType(void) const {
+        return "DER";
+    }
+};
+
+// This option enables/disables ALPN in the SSL handshake (if the SSL backend libcurl is built to
+// use supports it), which can be used to negotiate http2.
+class Alpn {
+  public:
+    Alpn(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+//  This option enables/disables NPN in the SSL handshake (if the SSL backend libcurl is built to
+//  use supports it), which can be used to negotiate http2.
+class Npn {
+  public:
+    Npn(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies that the server cert is for the server it is
+// known as.
+class VerifyHost {
+  public:
+    VerifyHost(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies the authenticity of the peer's certificate.
+class VerifyPeer {
+  public:
+    VerifyPeer(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+// This option determines whether libcurl verifies the status of the server cert using the
+// "Certificate Status Request" TLS extension (aka. OCSP stapling).
+class VerifyStatus {
+  public:
+    VerifyStatus(bool p_enabled = true) : enabled(p_enabled) {}
+
+    bool enabled;
+};
+
+struct TLSv1 {};
+#if LIBCURL_VERSION_MAJOR >= 7
```

Comment:
Wouldn't this disable SSLv2 and TLS for libcurl-8.0?

Reply:
* libcurl-7
Since libcurl-7 is over 10 years old this should not be a problem.

Reply:
I changed the macro to compare version base on the libcurl feature, it should solve the problem.

```cpp
#define __LIBCURL_VERSION_GTE(major, minor) \
    ((LIBCURL_VERSION_MAJOR > (major)) ||   \
     ((LIBCURL_VERSION_MAJOR == (major)) && (LIBCURL_VERSION_MINOR >= (minor))))

#ifndef SUPPORT_ALPN
#define SUPPORT_ALPN __LIBCURL_VERSION_GTE(7, 36)
#endif
```